projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.spec.ts

```diff
@@ -938,12 +941,67 @@ describe('IgxDateRangePicker', () => {
                     fixture.componentInstance.displayFormat = displayFormat;
                     fixture.detectChanges();
 
-                    expect(startInputEditor.inputFormat).toEqual(inputFormat);
+                    expect(startInputEditor.inputFormat).toEqual('MM/dd/yyyy');
                     expect(startInputEditor.displayFormat).toEqual(displayFormat);
-                    expect(endInputEditor.inputFormat).toEqual(inputFormat);
+                    expect(endInputEditor.inputFormat).toEqual('MM/dd/yyyy');
                     expect(endInputEditor.displayFormat).toEqual(displayFormat);
                 });
 
+                it('should set default inputFormat to the start/end editors with parts for day, month and year based on locale ', fakeAsync(() => {
+                    registerLocaleData(localeBg);
+                    registerLocaleData(localeJa);
+
+                    expect(fixture.componentInstance.inputFormat).toEqual(undefined);
+                    expect(dateRange.locale).toEqual('en-US');
+
+                    const startInputEditor = startInput.injector.get(IgxDateTimeEditorDirective);
+                    const endInputEditor = endInput.injector.get(IgxDateTimeEditorDirective);
+                    expect(startInputEditor.inputFormat).toEqual('MM/dd/yyyy');
+                    expect(endInputEditor.inputFormat).toEqual('MM/dd/yyyy');
+
+                    dateRange.locale = 'ja-JP';
+                    fixture.detectChanges();
+                    tick();
+
+                    expect(startInputEditor.inputFormat).toEqual('yyyy/MM/dd');
+                    expect(startInputEditor.nativeElement.placeholder).toEqual('yyyy/MM/dd');
+                    expect(endInputEditor.inputFormat).toEqual('yyyy/MM/dd');
+
+                    dateRange.locale = 'bg-BG';
+                    fixture.detectChanges();
+                    tick();
+
+                    expect(startInputEditor.inputFormat.normalize('NFKC')).toEqual('dd.MM.yyyy г.');
+                    expect(startInputEditor.nativeElement.placeholder.normalize('NFKC')).toEqual('dd.MM.yyyy г.');
+                    expect(endInputEditor.inputFormat.normalize('NFKC')).toEqual('dd.MM.yyyy г.');
+                }));
+
+                it('should resolve to the default locale-based inputFormat for the start/end editors in case the one set contains non-numeric time parts', fakeAsync(() => {
+                    registerLocaleData(localeBg);
+                    const startInputEditor = startInput.injector.get(IgxDateTimeEditorDirective);
+                    const endInputEditor = endInput.injector.get(IgxDateTimeEditorDirective);
+
+                    dateRange.locale = 'bg-BG';
+                    fixture.detectChanges();
+                    tick();
+
+                    dateRange.inputFormat = 'full';
```

Comment:
Still don't think this is intended at all. What this could do would be to allow the display format to affect to input format if not provided just like the grid, but that could be a separate target.